overlord/managers_test.go

```diff
@@ -0,0 +1,500 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package overlord_test
+
+// test the various managers and their operation together through overlord
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/partition"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	// XXX: here until we split out kernel_os
+	"github.com/snapcore/snapd/snappy"
+	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/systemd"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type mgrsSuite struct {
+	tempdir string
+
+	aa         *testutil.MockCmd
+	udev       *testutil.MockCmd
+	prevctlCmd func(...string) ([]byte, error)
+
+	o *overlord.Overlord
+}
+
+var _ = Suite(&mgrsSuite{})
+
+func (ms *mgrsSuite) SetUpTest(c *C) {
+	ms.tempdir = c.MkDir()
+	dirs.SetRootDir(ms.tempdir)
+
+	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "1")
+
+	// create a fake systemd environment
+	os.MkdirAll(filepath.Join(dirs.SnapServicesDir, "multi-user.target.wants"), 0755)
+
+	ms.prevctlCmd = systemd.SystemctlCmd
+	systemd.SystemctlCmd = func(cmd ...string) ([]byte, error) {
+		return []byte("ActiveState=inactive\n"), nil
+	}
+	ms.aa = testutil.MockCommand(c, "apparmor_parser", "")
+	ms.udev = testutil.MockCommand(c, "udevadm", "")
+
+	o, err := overlord.New()
+	c.Assert(err, IsNil)
+	ms.o = o
+}
+
+func (ms *mgrsSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("")
+	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "")
```

Comment:
seems I tend to forget that that exists, or maybe it wasn't there on day 0 ? anyway will fix